apps/mobile/src/screens/auth/components/sign-up-form/sign-up-form.tsx

```diff
@@ -1,53 +1,83 @@
 import React from "react";
 
-import { Button, Input, Link, Text, View } from "~/libs/components/components";
-import { RootScreenName } from "~/libs/enums/enums";
+import { Button, Input, Link, Text } from "~/libs/components/components";
+import { RootScreenName, UserValidationMessage } from "~/libs/enums/enums";
 import { useAppForm, useCallback } from "~/libs/hooks/hooks";
-import { userSignUpValidationSchema } from "~/packages/users/libs/validation-schemas/validation-schemas";
-import { type UserSignUpRequestDto } from "~/packages/users/users";
+import { globalStyles } from "~/libs/styles/styles";
+import { userSignUpValidationSchema } from "~/packages/users/users";
+import {
+	type UserSignUpRequestDto,
+	type UserSignUpSubmitDto,
+} from "~/packages/users/users";
 
 import { USER_SIGN_UP_DEFAULT_VALUES } from "./libs/constants";
 
 type Properties = {
 	onSubmit: (payload: UserSignUpRequestDto) => void;
 };
 const SignUpForm: React.FC<Properties> = ({ onSubmit }) => {
-	const { control, errors, handleSubmit } = useAppForm<UserSignUpRequestDto>({
-		defaultValues: USER_SIGN_UP_DEFAULT_VALUES,
-		validationSchema: userSignUpValidationSchema,
-	});
+	const { control, errors, handleSubmit, setError } =
+		useAppForm<UserSignUpSubmitDto>({
+			defaultValues: USER_SIGN_UP_DEFAULT_VALUES,
+			validationSchema: userSignUpValidationSchema,
+		});
 
 	const handleFormSubmit = useCallback((): void => {
-		void handleSubmit(onSubmit)();
-	}, [handleSubmit, onSubmit]);
+		void handleSubmit((data: UserSignUpSubmitDto) => {
+			const { confirmPassword, ...userData } = data;
+			if (confirmPassword === userData.password) {
+				onSubmit(userData);
+			} else {
+				setError("confirmPassword", {
+					message: UserValidationMessage.CONFIRM_PASSWORD_NOT_MATCH,
+					type: "manual",
+				});
```

Comment:
Create enum and move "confirmPassword" and "type" there

Reply:
Yes, it will be in the shared package. 
I think there should be two enums:
`const ConfirmField = {  PASSWORD: "confirmPassword" }` and `setErrorType = { CUSTOM: "custom"}`

Reply:
@fshabanov , which option could be better? For scaling it could be enum, I think. But for now may we prefer as @marharita08 suggested?

Reply:
I would create enum `CustomValidation = { field: 'field', errorType: 'errorType' } as const;`. I would also move the error message to that enum

Reply:
I have created enum CustomValidation in shared package in my branch.